docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+
+#### Stroy 3
+
+As a ZTP user, I want to safely move installed spoke clusters between hubs. Either manually (by applying CRs), or using a GitOps tool (e.g. ArgoCD). The spoke cluster on the target hub should behave the same and keep the status of BMHs.
+
+#### Story 4
+
+As an ACM/OADP or ZTP user, I want to restore/move a spoke cluster in non-installed status (draft/in-progress/etc). The cluster on target hub should, ideally, keep the same status. Or, at least, fail gracefully.
+
+### Implementation Details/Notes/Constraints
+
+#### Solution for BMHs reconcile (due to a missing status)
+
+According to BMO documentation, the suggested approach for moving BareMetalHost from one cluster to another is using the `baremetalhost.metal3.io/status` [annotation](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/bmo/status_annotation.md).
+I.e. Extract the current status of the BMH, store it in an annotation, create the BMH on the new hub:
+```
+# Save the status in json format to a file
+kubectl get bmh <name-of-bmh> -o jsonpath="{.status}" > status.json
+# Save the BMH and apply the status annotation to the saved BMH.
+kubectl -n metal3 annotate bmh <name-of-bmh> \
+  baremetalhost.metal3.io/status="$(cat status.json)" \
+  --dry-run=client -o yaml > bmh.yaml
+```
+This flow ensures that the BMH on the new hub has status (BMO applies the status using the annotation). Hence, as the BMH should be in state 'Provisioned', the BMH reconcile is avoided.
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+For BMHs in the active hub: (In assisted-service -> `bmh_agent_controller`)
+* Once the BMH is in 'Provisioned' state:
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
```

Comment:
Are we worried about a user creating an Agent CR manually? 

If we are, how do we prevent a user from adding an Agent CR that we never registered before? 


An idea I had was to label or annotate the Agent that it has been created by the service, that way we know it was an Agent that we created. This might not prevent the user from adding this themselves, but it's at least an extra layer that could prevent this scenario.

Reply:
I'm not sure it's worth the extra effort.
If someone is already creating the CR they'll just create whatever they need to.

In any case such an agent wouldn't be able to do anything. If we try to install with it it will stay disconnected unless an actual host matching this agent is booted with a discovery ISO.